src/agent/onefuzz-agent/src/tasks/fuzz/supervisor.rs

```diff
@@ -183,23 +182,19 @@ async fn start_supervisor(
         .reports_dir(&reports_dir)
         .setup_dir(&config.common.setup_dir)
         .job_id(&config.common.job_id)
-        .task_id(&config.common.task_id);
-
-    if let Some(tools) = &config.tools {
-        expand.tools_dir(&tools.path);
-    }
-
-    if let Some(target_exe) = &config.target_exe {
-        expand.target_exe(target_exe);
-    }
-
-    if let Some(target_options) = &config.target_options {
-        expand.target_options(target_options);
-    }
-
-    if let Some(input_marker) = &config.supervisor_input_marker {
-        expand.input_marker(input_marker);
-    }
+        .task_id(&config.common.task_id)
+        .set_optional_ref(&config.tools, |expand, tools| expand.tools_dir(&tools.path))
+        .set_optional_ref(&config.target_exe, |expand, target_exe| {
+            expand.target_exe(target_exe)
+        })
+        .set_optional_ref(&config.supervisor_input_marker, |expand, input_marker| {
+            expand.input_marker(input_marker)
+        });
+    let expand = if let Some(target_options) = &config.target_options {
```

Comment:
I see the use of `set_optional_ref`.  Can it be done here too?

Reply:
I needed to update the declareation of `set_optional_ref` to include a life time paramater. without it i was getting this error 
```
   --> onefuzz-agent\src\tasks\fuzz\supervisor.rs:193:76
    |
193 |         .set_optional_ref(&config.target_options, |expand, target_options| expand.target_options(target_options))
    |                                                    ------  --------------  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `target_options` escapes the closure body here
    |                                                    |       |
    |                                                    |       `target_options` is a reference that is only valid in the closure body
    |                                                    `expand` declared here, outside of the closure body

```